src/stages/models/manager.ts

```diff
@@ -197,6 +197,22 @@ export class StageManager {
       throw new StageNotFoundError(stagesErrorMessages.stageNotFound);
     }
 
+    const previousStageOrder = stage.order - 1;
+
+    // can't move to PENDING if previous stage is not COMPLETED
+    if (status === StageOperationStatus.PENDING && previousStageOrder > 0) {
+      const previousStage = await prisma.stage.findFirst({
+        where: {
+          jobId: stage.jobId,
+          order: previousStageOrder,
+        },
+      });
+
+      if (previousStage?.status !== StageOperationStatus.COMPLETED) {
+        throw new IllegalStageStatusTransitionError(`Previous stage is not ${StageOperationStatus.COMPLETED}`);
+      }
+    }
```

Comment:
As the `previousStageOrder` is coupled to the if statement and only to it, I would remove this empty line. But that's me

```suggestion
    const previousStageOrder = stage.order - 1;
    // can't move to PENDING if previous stage is not COMPLETED
    if (status === StageOperationStatus.PENDING && previousStageOrder > 0) {
      const previousStage = await prisma.stage.findFirst({
        where: {
          jobId: stage.jobId,
          order: previousStageOrder,
        },
      });

      if (previousStage?.status !== StageOperationStatus.COMPLETED) {
        throw new IllegalStageStatusTransitionError(`Previous stage is not ${StageOperationStatus.COMPLETED}`);
      }
    }
```